framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -63,4 +86,99 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 			await this._sidechainAPIClient.disconnect();
 		}
 	}
+	public async getCertificateFromAggregateCommit(
+		aggregateCommit: AggregateCommit,
+	): Promise<Certificate> {
+		const block = await this._mainchainAPIClient.invoke<Block>('chain_getBlockByHeight', {
+			height: aggregateCommit.height,
+		});
+
+		const certificate = computeCertificateFromBlockHeader(block.header);
+		certificate.aggregationBits = block.header.aggregateCommit.aggregationBits;
+		certificate.signature = aggregateCommit.certificateSignature;
+
+		return certificate;
+	}
+
+	public async getNextCertificateFromAggregateCommits(
+		lastCertifiedHeight: number,
+		aggregateCommits: AggregateCommit[],
+	): Promise<Certificate | null> {
+		const block = await this._mainchainAPIClient.invoke<Block>('chain_getBlockByHeight', {
```

Comment:
should this be _sidechainAPIClient?

Reply:
this should be `_sidechainAPIClient `, if it doesn't exist then `this._apiClient` same as above